### PR TITLE
Fix #402: parse for error

### LINKED ISSUE #402: parse: RuntimeWarning: coroutine '<name>' was never awaited
pkcli.test should parse the error log for this error, and fail the test.

### PR DESCRIPTION
**Pull request by gurhar1133**

### CODE REVIEW

File: tests/pkcli/warning_data/tests/1_test.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+u"""fails due to RuntimeWarning
+
+:copyright: Copyright (c) 2019 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from __future__ import absolute_import, division, print_function
```
Comment (robnagler) [inline tests/pkcli/warning_data/tests/1_test.py:7]: `__future__` unnecessary

File: tests/pkcli/warning_data/tests/1_test.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+u"""fails due to RuntimeWarning
+
+:copyright: Copyright (c) 2019 RadiaSoft LLC.  All Rights Reserved.
```
Comment (robnagler) [inline tests/pkcli/warning_data/tests/1_test.py:4]: 2023

File: tests/pkcli/warning_test.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+"""RuntimeWarning test
+
+:copyright: Copyright (c) 2019 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+import pytest
```
Comment (robnagler) [inline tests/pkcli/warning_test.py:7]: unnecessary import

File: tests/pkcli/warning_test.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+"""RuntimeWarning test
+
+:copyright: Copyright (c) 2019 RadiaSoft LLC.  All Rights Reserved.
```
Comment (robnagler) [inline tests/pkcli/warning_test.py:4]: 2023

File: pykern/pkcli/test.py [hunk 5]
```diff
@@ -165,8 +167,14 @@ def _except(exc, output, restartable):
                     # 2 means KeyboardInterrupt
                     # POSIT: pkunit.restart_or_fail uses this
                     return "restart"
+            m = "FAIL"
             self.failures.append(output)
-            return f"FAIL {output}"
+            if _runtime_warning(output):
```
Comment (robnagler) [inline pykern/pkcli/test.py:172]: This is unnecessary.

File: pykern/pkcli/test.py [hunk 6]
```diff
@@ -18,6 +18,8 @@
 SUITE_D = "tests"
 
 _TEST_PY = re.compile(r"_test\.py$")
+_RUNTIME_WARNING_FAIL = r"RuntimeWarning: coroutine .+ was never awaited"
```
Comment (robnagler) [inline pykern/pkcli/test.py:21]: `r` is unnecessary. The name is `_COROUTINE_NEVER_AWAITED`. It should be a `re.compile`.

File: pykern/pkcli/test.py [hunk 7]
```diff
@@ -165,8 +167,14 @@ def _except(exc, output, restartable):
...
                     # POSIT: pkunit.restart_or_fail uses this
                     return "restart"
+            m = "FAIL"
             self.failures.append(output)
-            return f"FAIL {output}"
+            if _runtime_warning(output):
+                m = "Invalid RuntimeWarning Fail"
+            return m + f" {output}"
+
+        def _runtime_warning(output):
```
Comment (robnagler) [inline pykern/pkcli/test.py:176]: `_coroutine_never_awaited`

File: pykern/pkcli/test.py [hunk 8]
```diff
@@ -177,6 +185,8 @@ def _try(output, restartable):
                     output=output,
                     env=_env(restartable),
                 )
+                if _runtime_warning(output):
+                    raise AssertionError("coroutine not awaited, warning raised")
```
Comment (robnagler) [inline pykern/pkcli/test.py:189]: I think what should happen is that we should append the FAILURE description to the end of output. We could actually edit the pytest output:

`======================== 3 passed, 2 warnings in 27.88s ========================`

We don't know what failed or passed here, but we can simply replace this line with something like:

```
FAILED: Detected asyncio problem: coroutine was never awaited
```

File: tests/pkcli/warning_test.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (robnagler) [inline tests/pkcli/warning_test.py:1]: This should be in test_data/tests/3_test.py. It should then have a log. It should actually be an asyncio example:

```
def test_coroutine_never_awaited():
    async def _test():
        asyncio.sleep(0)
    asyncio.run(_test)
```

Something like that.

File: tests/pkcli/warning_test.py [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+"""RuntimeWarning test
+
+:copyright: Copyright (c) 2019 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+import pytest
+
+
+def test_warning(capsys):
```
Comment (robnagler) [inline tests/pkcli/warning_test.py:10]: This is what pkcli/test_test.py does.

**Review by robnagler — CHANGES_REQUESTED** [review]

**Comment by robnagler** [comment]
@gurhar1133 please see my changes an lmk.

**Comment by gurhar1133** [comment]
Yes, moving the test to a case dir and using capsys simplifies pkcli/test.py a lot. I removed the unused _sub_output function. lmk if you want me to make more changes

**Review by robnagler — APPROVED** [review]

**Pull request merged**